src/vaults/BalanceSheet.sol

```diff
@@ -148,11 +146,19 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
     /// @dev This function is mostly useful to keep higher level integrations CEI adherent.
     function noteRevoke(PoolId poolId, ShareClassId scId, address from, uint128 shares)
         external
-        authOrManager(poolId, scId)
+        authOrManager(poolId)
     {
         _noteRevoke(poolId, scId, from, shares);
     }
 
+    function setPricePoolPerAsset(PoolId poolId, ShareClassId scId, D18 value) external authOrManager(poolId) {
+        TransientStorageLib.tstore(keccak256(abi.encode("pricePoolPerAsset", poolId, scId)), value.raw());
```

Comment:
Should we maybe include the assetID here? Not sure if otherwise would be quite easy to override here.